ouroboros-consensus/src/Ouroboros/Consensus/Ledger/Query.hs

```diff
@@ -182,19 +260,29 @@ instance SameDepIndex (BlockQuery blk) => SameDepIndex (Query blk) where
     = Just Refl
   sameDepIndex GetSystemStart _
     = Nothing
+  sameDepIndex GetChainBlockNo GetChainBlockNo
+    = Just Refl
+  sameDepIndex GetChainBlockNo _
+    = Nothing
+  sameDepIndex GetChainPoint GetChainPoint
+    = Just Refl
+  sameDepIndex GetChainPoint _
+    = Nothing
```

Comment:
Would this also work?

```haskell
  sameDepIndex (BlockQuery blockQueryA) (BlockQuery blockQueryB) = Just Refl
  sameDepIndex GetSystemStart                   GetSystemStart                 = Just Refl
  sameDepIndex GetChainBlockNo                GetChainBlockNo             = Just Refl
  sameDepIndex GetChainPoint                     GetChainPoint                   = Just Refl
  sameDepIndex _                                          _                                         = Nothing 
```
(I don't know what GH does with the alignment, sorry)

My guess is that it would, but we want to prevent a situation in which we add another constructor and it gets caught by the wildcard case.

Reply:
I suspect being able to get a compile error is more important.

Reply:
Yeah, definitely!